### PR TITLE
refactor: use Stack class for control stack in parser

### PR DESCRIPTION
**Pull request by gumb0**
Required for #312

### CODE REVIEW

File: lib/fizzy/stack.hpp [hunk 1]
```diff
@@ -28,6 +28,12 @@ class Stack : public std::vector<T>
 
     void push(T val) { emplace_back(val); }
 
+    template <typename... Args>
+    void emplace(Args&&... args)
```
Comment (axic) [inline lib/fizzy/stack.hpp:32]: Should add a test case for this.

File: lib/fizzy/parser_expr.cpp [hunk 2]
```diff
@@ -120,7 +121,7 @@ parser_result<Code> parse_expr(const uint8_t* pos, const uint8_t* end, const Mod
     // The stack of control frames allowing to distinguish between block/if/else and label
     // instructions as defined in Wasm Validation Algorithm.
     // For a block/if/else instruction the value is the block/if/else's immediate offset.
-    std::stack<ControlFrame> control_stack;
+    Stack<ControlFrame> control_stack;
```
Comment (axic) [inline lib/fizzy/parser_expr.cpp:124]: Does this makes any measurable difference in speed?

Comment (chfast) [reply lib/fizzy/parser_expr.cpp:123]: This uses `std::vector` instead of `std::deque` now, but it seems make no difference (see below).

**Comment by chfast** [comment]
Benchmarked this on rather noisy could instance.
```
fizzy/parse/blake2b_mean                             +0.0342         +0.0341            17            17            17            17
fizzy/parse/ecpairing_mean                           -0.0104         -0.0104           932           922           932           922
fizzy/parse/keccak256_mean                           -0.0069         -0.0069            27            27            27            27
fizzy/parse/memset_mean                              -0.0159         -0.0160             4             4             4             4
fizzy/parse/mul256_opt0_mean                         -0.0166         -0.0166             5             5             5             5
fizzy/parse/sha1_mean                                +0.0078         +0.0078            26            26            26            26
fizzy/parse/sha256_mean                              +0.0364         +0.0364            41            43            41            43
fizzy/parse/micro/factorial_mean                     -0.0099         -0.0100             1             1             1             1
fizzy/parse/micro/fibonacci_mean                     -0.0123         -0.0123             1             1             1             1
fizzy/parse/micro/host_adler32_mean                  -0.0132         -0.0132             2             2             2             2
fizzy/parse/micro/spinner_mean                       +0.0185         +0.0184             1             1             1             1
```

**Pull request merged**